modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc

```diff
@@ -0,0 +1,361 @@
+  }
+
+  drupal_set_message(t('The denomination type @type has been successfully deleted.', array(
+    '@type' => $form_state['denomination_type_info']->currency_code
+  )));
+  drupal_goto('admin/commerce/config/pos/denomination');
+}
+
+function commerce_pos_currency_view_denominations($currency_id) {
+  $output = '';
+
+  // Get the denomination type info.
+  $denomination_type_info = commerce_pos_denomination_get_currency_types_from_id(array($currency_id));
+
+  if (!empty($denomination_type_info)) {
+    $denomination_type_info = $denomination_type_info[$currency_id];
+
+    $output .= l(t('Add Denomination'), 'admin/commerce/config/pos/denomination-type/' . $denomination_type_info->currency_id . '/denomination/add', array(
+      'attributes' => array('class' => array('add-denomination', 'button'))
+    ));
+
+    // Now, get the denominations for this denomination type.
+    $denominations = commerce_pos_currency_denomination_get_denominations(array($denomination_type_info->currency_code));
+    if (!empty($denominations)) {
+      $output .= t('<h3>Current Denominations for Type @type</h3>', array(
+        '@type' => $denomination_type_info->currency_code
+      ));
+
+      $output .= '<table class="commerce-pos-current-denominations">';
+      $output .= '<tr>';
+      $output .= '<th>' . t('Title') . '</th>';
+      $output .= '<th>' . t('Amount') . '</th>';
+      $output .= '<th>' . t('Options') . '</th>';
+      $output .= '</tr>';
+
+      foreach ($denominations[$denomination_type_info->currency_id] as $denomination) {
+        $output .= '<tr>';
+        $output .= '<td>' . $denomination->denomination_title . '</td>';
+        $output .= '<td>' . $denomination->denomination_amount . '</td>';
+        $output .= '<td>' . l(t('Delete Denomination'), 'admin/commerce/config/pos/denomination-type/' . $currency_id . '/denomination/' . $denomination->denomination_id . '/delete') . '</td>';
+        $output .= '</tr>';
+      }
+
+      $output .= '</table>';
+    }
+    else {
+      $output .= t('<p>Currently, no denominations have been added for this denomination type.</p>');
+    }
+  }
+  else {
+    $output .= t('<p>Invalid denomination type.</p>');
+  }
+
+  return $output;
+}
+
+/**
+ * The form for adding a new denomination.
+ */
+function commerce_pos_currency_add_denomination_form($form, &$form_state, $currency_id) {
+  $form_state['currency_id'] = $currency_id;
+
+  // Get the denomination info for this type.
+  $form_state['denomination_type_info'] = commerce_pos_denomination_get_currency_types_from_id(array($currency_id));
+
+  if (!empty($form_state['denomination_type_info'])) {
+    $form_state['denomination_type_info'] = $form_state['denomination_type_info'][$currency_id];
+
+    drupal_set_title(t('Add Denomination for Type @type', array(
+      '@type' => $form_state['denomination_type_info']->currency_code
+    )));
+
+    $form['commerce_pos_denomination_name'] = array(
+      '#type' => 'textfield',

```

Comment:
You can use 'machine_name' type for this and link it to the title field to make this a better UX: https://api.drupal.org/api/drupal/developer%21topics%21forms_api_reference.html/7.x#machine_name